homeassistant/components/nest/camera_sdm.py

```diff
@@ -203,13 +203,11 @@ async def async_camera_image(
         self, width: int | None = None, height: int | None = None
     ) -> bytes | None:
         """Return bytes of camera image."""
-        # Use the thumbnail from RTSP stream
+        # Use the thumbnail from RTSP stream, or a placeholder if stream is
+        # not supported (e.g. WebRTC)
         stream = await self.async_create_stream()
         if stream:
             return await stream.async_get_image(width, height)
-        if self.frontend_stream_type != STREAM_TYPE_WEB_RTC:
-            return None
-        # Use a placeholder image for WebRTC cameras with no thumbnail
         if not self._placeholder_image:
             self._placeholder_image = await self.hass.async_add_executor_job(
```

Comment:
Is the placeholder_image loaded like this instead of in __init__ because it's not used much and it doesn't make sense to keep it in memory? Or can it change over time?

Reply:
It used to compute it on the fly using an image library, so maybe it was more important then, but now it still does a little disk I/o reading the image from disk so not sure if init can be async. This is currently loaded once per object instance, but it doesn't need to be, it can just be loaded once per server. If you have a specific suggestion for improvement i'll take it.

Reply:
Got it. It's not too bad since it's only loaded once, but moving it to init would avoid having to check whether it exists each time.
I think you can use `hass.add_job` from init, and as long as the function is not decorated with `@callback` it will run in an executor.

Reply:
The current way is fine by me too. Keeping it lazy will avoid loading the placeholder in cases where it's not used.

Reply:
OK, how about we go with this (as it already is) and I will send a follow up to simplify. I think something like `@cached_property` or something else could maybe simplify this.

Reply:
Sure.
I haven't used `@cached_property` before if we use it, one other (separate) thing we can do is consider using the ttl feature to clear out old images. I'm not sure what the best timeout would be though - seeing an hours old image is misleading but at least you might know it is old from the time of day. Having a minutes old image might be even more misleading because you might not know it's a stale image, and any non fixed objects in the frame might have already changed.

Reply:
We're just talking about a [placeholder](https://github.com/home-assistant/core/blob/dev/homeassistant/components/nest/placeholder.png) which is a black image with a white spot, so no freshness concerns.